collector/src/main/java/com/bitmovin/analytics/data/Backend.kt

```diff
@@ -0,0 +1,13 @@
+package com.bitmovin.analytics.data
+
+import com.bitmovin.analytics.retryBackend.RetryCallback
+
+interface Backend {
+    fun send(eventData: EventData)
+    fun sendAd(eventData: AdEventData)
+}
+
+interface CallbackBackend {
```

Comment:
can you move this into a separate file?
Any reason you name this `RetryCallback`? This is just the normal callback as I can see.
What we can do though, is wrapping the `okhttp.Callback` into our own class (and then using some interface like the `RetryCallback`, just with a better naming as it's not related to `Retry`) - so we could decouple it from the `okhttp` (I think Daniel has mentioned that somewhere).
So you would have an interface like this e.g. (for now, we don't even need the success-path):
```
fun onFailure(e: Exception, cancel: () => Unit)
```
as those are the only 2 things we are interested in in the `RetryBackend`.

Reply:
Moved to separate file, and changed interface in https://github.com/bitmovin/bitmovin-analytics-collector-android/pull/132/commits/f9d6c68a0c58cc6478feeacead3ae501992732d7